tests/expose_test.go

```diff
@@ -484,10 +864,57 @@ var _ = Describe("[rfe_id:253][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 
 				By("Starting a job which tries to reach the VMI via the ClusterIP service.")
 				job = runHelloWorldJob(serviceIP, servicePort, vm.Namespace)
+				jobCleanupFuncs = append(jobCleanupFuncs, cleanupJob(job.GetName(), job.GetNamespace()))
 
 				By("Waiting for the job to report a failed connection attempt.")
 				Expect(tests.WaitForJobToFail(job, 120*time.Second)).To(Succeed())
-			})
+
+				// this way, the test does not attempt to stop the VM in the `AfterEach` section
+				vm = nil
```

Comment:
maybe better to use defer, so even if there is a panic it would happen

Reply:
right, so other question, having it in mind.
you create a new stopped vm for each test, right ?
so why to not stop / delete it on each end of a test ? 

generally speaking, you dont delete it , just stop it if all goes well, is it ok ?

Reply:
That means I need to create the VM once, and start / stop on the setup / teardown. I'd prefer to move as far away from `BeforeAll` as I can. 

Plus, creating the VM object is peanuts when compared to starting it.

Reply:
if all goes well, who will stop the vm ?
each test would put nil and wont stop , nor delete it, or i miss something?

i find this reuse, even that it save time, little dangerous, and cause the code bit harder to maintain, we dont do it usually, this is my opinion at least.

Reply:
> if all goes well, who will stop the vm ?

If all goes well, the VM is deleted *within* the test. There's no VM to stop. 

> each test would put nil and wont stop , nor delete it, or i miss something?
> 
> i find this reuse, even that it save time, dangerous, and cause the code bit harder to maintain, we dont do it usually, this is my opinion at least.

> moreover, in case of a error in one of the test, you still create a new stopped vm, and there will be a leak of the previous vm, (it would be a stopped one, but still) right ?

Here you're right; error or not, the VM object is leaking. Gotta delete it.

Reply:
Done.